test/metric/dimension/provider.go

```diff
@@ -17,11 +17,11 @@ type ExpectedDimensionValue struct {
 }
 
 func (d *ExpectedDimensionValue) IsKnown() bool {
-	return d.Value == nil
+	return d.Value != nil
```

Comment:
Nit: I think you may need to rebase on main. I think this fix is already in.

Reply:
yes will have to rebase. Somebody told me it wasn't fixed yet & my repo was failing even with newest commits when I was doing this :)